shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java

```diff
@@ -216,6 +221,104 @@ private BindingTableRule createBindingTableRule(final String bindingTableGroup)
+        }
+        checkSameAlgorithmOnDatabase(sampleTableRule, tableRule, sampleTableRule.getActualDatasourceNames().stream().findFirst().get(), bindingTableGroup);
+        for (String each : sampleTableRule.getActualDatasourceNames()) {
+            Collection<String> sampleActualTableNames = sampleTableRule.getActualTableNames(each).stream().map(optional -> substring(optional)[1])
+                    .filter(optional -> !optional.isEmpty()).collect(Collectors.toList());
+            Collection<String> actualTableNames = tableRule.getActualTableNames(each).stream().map(optional -> substring(optional)[1])
+                    .filter(optional -> !optional.isEmpty()).collect(Collectors.toList());
+            if (!CollectionUtils.isEqualCollection(sampleActualTableNames, actualTableNames)) {
+                throw new ShardingSphereConfigurationException("The %s on bindingTableGroup `%s` are inconsistent", "actualTableNames", bindingTableGroup);
+            }
+            checkSameAlgorithmOnTable(sampleTableRule, sampleTableRule.getActualTableNames(each).stream().findFirst().get(), tableRule,
+                    tableRule.getActualTableNames(each).stream().findFirst().get(), bindingTableGroup);
+        }
+    }
+    
+    private void checkSameAlgorithmOnDatabase(final TableRule sampleTableRule, final TableRule tableRule, final String dataSourceName,
+                                              final String bindingTableGroup) {
+        Collection<String[]> algorithmExpressions = new ArrayList<>();
+        String logicName = substring(dataSourceName)[0];
+        algorithmExpressions.add(new String[] {getAlgorithmExpression(sampleTableRule, true), logicName, getShardingColumn(sampleTableRule.getDatabaseShardingStrategyConfig())});
+        algorithmExpressions.add(new String[] {getAlgorithmExpression(tableRule, true), logicName, getShardingColumn(tableRule.getDatabaseShardingStrategyConfig())});
+        checkSameAlgorithmExpression(algorithmExpressions, "databaseShardingStrategyConfig", bindingTableGroup);
+    }
+    
+    private String getAlgorithmExpression(final TableRule tableRule, final boolean databaseOrTable) {
+        if (null == tableRule.getDatabaseShardingStrategyConfig() || null == tableRule.getTableShardingStrategyConfig()) {
+            return "";
+        }
+        String shardingAlgorithmName = databaseOrTable ? tableRule.getDatabaseShardingStrategyConfig().getShardingAlgorithmName()
+                : tableRule.getTableShardingStrategyConfig().getShardingAlgorithmName();
+        ShardingAlgorithm shardingAlgorithm = shardingAlgorithms.get(shardingAlgorithmName);
+        return null == shardingAlgorithm ? "" : checkWithDefaultValue(shardingAlgorithm.getProps().getProperty("algorithm-expression"), "");
+    }
+    
+    private String checkWithDefaultValue(final String str, final String defaultValue) {
```

Comment:
@cheese8 Can we remove this method with `StringUtils.defaultString()`?

Reply:
OK, it's better